public/app/features/search/components/SearchCard.tsx

```diff
@@ -64,7 +64,11 @@ export function SearchCard({ editable, item, onTagSelected, onToggleChecked }: P
     if (item.uid && !lastUpdated) {
       const dashboard = await backendSrv.getDashboardByUid(item.uid);
       const { updated } = dashboard.meta;
-      setLastUpdated(new Date(updated).toLocaleString());
+      if (updated) {
```

Comment:
Yeah, confused where this change to search comes from

Reply:
I couldn't find definitive info regarding wether `updated` can be undefined, but according to https://github.com/grafana/grafana/pull/47083/files#diff-a59fe074f33f72d2359deef1d8321eeef3bcfb19d2471f25cc0025817dcbce0eR426 and https://github.com/grafana/grafana/blob/b9ab971b3fa2b3086e28c02cb4e597cb82bc6f1d/public/app/types/dashboard.ts#L12 it can be undefined. the changes in https://github.com/grafana/grafana/pull/47083/files#diff-a59fe074f33f72d2359deef1d8321eeef3bcfb19d2471f25cc0025817dcbce0eR426 makes `getDashboardbyUid` return `DashboardDTO` instead of `any`, and `Date` constructor can't accept `undefined`. i'm also ok with doing `      setLastUpdated(new Date(updated!).toLocaleString());` but that felt a bit wrong.


in SearchCard, when expanded, if `lastUpdated` is undefined a spinner is shown indicating a loading state, which assumes `dashboard.meta.updated` is always defined (but does not match the TS types).

to be on the safe side, `lastUpdated` is now of type `string | null` where null indicates "no info" and the relevant portion in the card details is not rendered: 

![Screenshot 2022-03-30 at 10 00 56](https://user-images.githubusercontent.com/1170767/160795283-736d6972-1338-46aa-b8db-145c73d304a4.png)
![Screenshot 2022-03-30 at 10 02 21](https://user-images.githubusercontent.com/1170767/160795289-f360e5ec-973d-43f4-b4b5-ce4b1a12997d.png)